httpclient5/src/main/java/org/apache/hc/client5/http/impl/nio/H2SharingConnPool.java

```diff
@@ -288,20 +290,14 @@ PoolEntry<T, C> lease() {
         long release(final PoolEntry<T, C> entry, final boolean reusable) {
             lock.lock();
             try {
```

Comment:
@arturobernalg Could you please add this check here?
```
if (!reusable) {
     entry.discardConnection(CloseMode.GRACEFUL);
}
```

Reply:
@ok2c  done